src/components/Slider/Slider.js

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import "./Slider.css";
+import SliderNavButton from "./SliderNavButton";
+import FeaturedBanners from "../../mocks/en-us/featured-banners.json";
+
+export default class Slider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      slideIndex: 1,
+      dataSlider: FeaturedBanners.results,
+    };
+  }
+
+  nextSlide = () => {
+    if (this.state.slideIndex !== this.state.dataSlider.length) {
+      this.setState({ slideIndex: this.state.slideIndex + 1 });
+    } else if (this.state.slideIndex === this.state.dataSlider.length) {
+      this.setState({ slideIndex: 1 });
+    }
+  };
+
+  prevSlide = () => {
+    if (this.state.slideIndex !== 1) {
+      this.setState({ slideIndex: this.state.slideIndex - 1 });
+    } else if (this.state.slideIndex === 1) {
+      this.setState({ slideIndex: this.state.dataSlider.length });
+    }
+  };
```

Comment:
Una ventaja de usar functional components es que estas funciones verían su código muy reducido porque ya no tenemos que estar usando `this.state` al usar el hook useState.